aria/appilcation_tools.py

```diff
@@ -0,0 +1,257 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+import json
+import shutil
+import tarfile
+import tempfile
+from copy import copy
+from datetime import datetime
+
+from aria.logger import LoggerMixin
+from aria.exceptions import StorageError
+
+
+class StorageManager(LoggerMixin):
+    def __init__(
+            self,
+            model_storage,
+            resource_storage,
+            blueprint_path,
+            blueprint_id,
+            blueprint_plan,
+            deployment_id,
+            deployment_plan,
+            *args,
+            **kwargs):
+        super(StorageManager, self).__init__(*args, **kwargs)
+        self.model_storage = model_storage
+        self.resource_storage = resource_storage
+        self.blueprint_path = blueprint_path
+        self.blueprint_id = blueprint_id
+        self.blueprint_plan = blueprint_plan
+        self.deployment_id = deployment_id
+        self.deployment_plan = deployment_plan
+
+    @classmethod
+    def from_blueprint(
+            cls,
+            model_storage,
+            resource_storage,
+            blueprint_path,
+            blueprint_id):
+        return cls(
+            model_storage,
+            resource_storage,
+            blueprint_path,
+            blueprint_id,
+            deployment_id=None,
+            deployment_plan=None)
+
+    def create_blueprint_storage(self, source, main_file_name=None):
+        """
+        create blueprint model & resource
+        """
+        assert self.blueprint_path and self.blueprint_id
+        assert hasattr(self.resource_storage, 'blueprint')
+        assert hasattr(self.model_storage, 'blueprint')
+
+        self.logger.debug('creating blueprint resource storage entry')
+        self.resource_storage.blueprint.upload(
+            entry_id=self.blueprint_id,
+            source=os.path.dirname(source))
+        self.logger.debug('created blueprint resource storage entry')
+
+        self.logger.debug('creating blueprint model storage entry')
+        now = datetime.now()
+        blueprint = self.model_storage.blueprint.model_cls(
+            plan=self.blueprint_plan,
+            id=self.blueprint_id,
+            description=self.blueprint_plan.get('description'),
+            created_at=now,
+            updated_at=now,
+            main_file_name=main_file_name or 'blueprint.yaml'
+        )
+        self.model_storage.blueprint.store(blueprint)
+        self.logger.debug('created blueprint model storage entry')
+
+    def create_nodes_storage(self):
+        """
+        create nodes model
+        """
+        assert hasattr(self.model_storage, 'node')
+        assert hasattr(self.model_storage, 'relationship')
+
+        for node in copy(self.blueprint_plan['nodes']):
```

Comment:
for node in self.blueprint_plan['nodes']:
    node_dict = node.copy()
    ...
    node_model = self.model_storage.node.model_cls(